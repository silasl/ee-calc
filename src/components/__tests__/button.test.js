import React from 'react';
import { shallow } from 'enzyme';

import Button from '../button';

describe(Button, () => {
    const props = {
        handleClick: () => {},
        value: 'A',
    }

    it('should render correctly', () => {
        expect(shallow(<Button {...props} />)).toMatchSnapshot();
    });

    it('should render correctly with a label', () => {
        expect(shallow(<Button {...props} label="B" />)).toMatchSnapshot();
    });

    it('should render correctly with a className', () => {
        expect(shallow(<Button {...props} className="foo" />)).toMatchSnapshot();
    });
});
