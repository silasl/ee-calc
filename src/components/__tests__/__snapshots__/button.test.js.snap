// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<styled.button
  onClick={[Function]}
>
  A
</styled.button>
`;

exports[`Button should render correctly with a className 1`] = `
<styled.button
  className="foo"
  onClick={[Function]}
>
  A
</styled.button>
`;

exports[`Button should render correctly with a label 1`] = `
<styled.button
  onClick={[Function]}
>
  B
</styled.button>
`;
